Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js

```diff
@@ -681,11 +681,29 @@ WI.DOMManager = class DOMManager extends WI.Object
             return;
 
         let target = WI.assumingMainTarget();
+
         let commandArguments = {
             enabled,
             highlightConfig: DOMManager.buildHighlightConfig(),
             showRulers: WI.settings.showRulersDuringElementSelection.value,
         };
+        if (WI.settings.showGridOverlayDuringElementSelection.value) {
+            commandArguments.gridOverlayConfig = {
+                gridColor: WI.DOMNode.defaultLayoutOverlayColor.toProtocol(),
```

Comment:
I wonder if it will be confusing that we use one of the five default colors, effectively meaning there is a 4/5 chance that the grid/flex color won't match the one in the Layout sidebar. That said, we can't really resolve the color for any given node here because we don't know what node someone will hover. Have you considered perhaps using grey color to give a sense of the overlay without always using magenta? Maybe some screenshots would convince me this is fine? (I'm lacking spare cycles to build this patch myself right now 😬)

Reply:
that's a fair point.  i figured that if the magenta was good enough to be our initial option, it'd be good enough for any node 😅

i tried a few other colors like grey, but they all felt a bit too muted and not really very "stand out", whereas the magenta is a super nice contrast with the blue color we use for content (and green for padding)

while i do agree that maybe using the same color that could match with an already enabled/showing overlay could be a bit confusing for the user, if it's ok with you i'd rather handle that problem once this functionality is landed as changing the color in the frontend is _way_ easier than modifying the entire backend :)

Reply:
After looking at this, I think you are right that magenta is a good color here given the other colors in use. Thank you for explaining your reasoning here!